src/QuadForm/Spaces.jl

```diff
@@ -702,6 +702,33 @@ function orthogonal_sum(V::HermSpace, W::HermSpace)
   return VplusW, f1, f2
 end
 
+function _orthogonal_sum_with_injections_and_projections(x::Vector{<:QuadSpace})
+  @assert length(x) >= 2
+  K = base_ring(x[1])
+  @assert all(i -> base_ring(x[i]) === K, 2:length(x))
```

Comment:
```suggestion
  @req all(i -> base_ring(x[i]) === K, 2:length(x)) "lattices not defined over the same field."
```

Reply:
Do you want I make all the `@assert` into `@req` ? Since they were supposedly internal functions I just wanted to make sure we put the write input. But if you want I can make them in all the functions as `@req`.

Reply:
Apparently `@assert` can get optimized away by the compiler. For input checking `@req` is probably better.